### PR TITLE
[Core] Linter rule for service_name.json

### PR DESCRIPTION
**Pull request by qwordy**
Check format of service_name.json. Command and AzureServiceName are required. Others are optional.
Each highest level command group should have reference in service_name.json.
It only runs on modified extensions. I have tested it with modification on extension. This PR itself will not modify any extension.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

File: src/service_name.json [hunk 1]
```diff
@@ -0,0 +1,342 @@
...
+  {
+    "Command": "az communication",
+    "AzureServiceName": "Azure Communication Service",
+    "URL": "https://docs.microsoft.com/azure/communication-services/"
+  },
+  {
+    "Command": "az confluent",
+    "AzureServiceName": "None",
+    "URL": ""
+  },
```
Comment (fengzhou-msft) [inline src/service_name.json:61]: This is a third-party service in LiftR. Do we just use `None` for the `AzureServiceName`? Will it cuase problems if there are mulitple `None` `AzureServiceName`?

File: src/service_name.json [hunk 2]
```diff
@@ -0,0 +1,342 @@
...
+    "URL": "https://docs.microsoft.com/azure/migrate/"
+  },
+  {
+    "Command": "az peering",
+    "AzureServiceName": "Azure Network",
+    "URL": "https://docs.microsoft.com/azure/virtual-network/virtual-network-peering-overview"
+  },
+  {
+    "Command": "az portal",
+    "AzureServiceName": "Azure portal",
```
Comment (fengzhou-msft) [inline src/service_name.json:259]: Should it be <code>Azure **P**ortal</code>?

File: src/service_name.json [hunk 3]
```diff
@@ -0,0 +1,342 @@
...
+  },
+  {
+    "Command": "az quantum",
+    "AzureServiceName": "Azure Quantum",
+    "URL": "https://docs.microsoft.com/azure/quantum/"
+  },
+  {
+    "Command": "az resource-mover",
+    "AzureServiceName": "Azure CLI",
+    "URL": "https://docs.microsoft.com/cli/azure/"
```
Comment (fengzhou-msft) [inline src/service_name.json:285]: I think this is a service. @houk-ms to shed some light.

File: scripts/ci/service_name.py [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+
+def _store_parsers(parser, parser_keys, parser_values, sub_parser_keys, sub_parser_values):
+    for s in parser.subparsers.values():
+        parser_keys.append(_get_parser_name(s))
+        parser_values.append(s)
+        if _is_group(s):
+            for c in s.choices.values():
+                sub_parser_keys.append(_get_parser_name(c))
+                sub_parser_values.append(c)
+                _store_parsers(c, parser_keys, parser_values, sub_parser_keys, sub_parser_values)
```
[the remark this answers is not in the document]
Comment (qwordy) [reply scripts/ci/service_name.py:77]: Reuse code from azure-cli-extensions

File: src/service_name.json (hunk 1 above)
Comment (qwordy) [reply src/service_name.json]: Updated

File: scripts/ci/service_name.py [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+                   commands_loader_cls=MainCommandsLoader,
+                   invocation_cls=AzCliCommandInvoker,
+                   parser_cls=AzCliCommandParser,
+                   help_cls=AzCliHelp)
+    help_files = get_extension_help_files(az_cli)
+    high_command_set = set()
+    for help_file in help_files:
+        if help_file.command:
+            high_command_set.add(help_file.command.split()[0])
+    print('high_command_set:')
```
Comment (fengzhou-msft) [inline scripts/ci/service_name.py:100]: Is `high command` an existing term or coined by you? Is it OK to use `command group` to refer to the same thing?

**Review by fengzhou-msft — APPROVED** [review]

File: scripts/ci/service_name.py (hunk 5 above)
Comment (qwordy) [reply scripts/ci/service_name.py:100]: The left most word in a command, e.g., `vm`, `disk`. Let me add a comment.

File: scripts/ci/service_name.py [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+
+
+def check():
+    az_cli = AzCli(cli_name='az',
+                   commands_loader_cls=MainCommandsLoader,
+                   invocation_cls=AzCliCommandInvoker,
+                   parser_cls=AzCliCommandParser,
+                   help_cls=AzCliHelp)
+    help_files = get_extension_help_files(az_cli)
+    high_command_set = set()
```
Comment (qwordy) [inline scripts/ci/service_name.py:96]:
```suggestion
    # High command represents left most word in a command, e.g., vm, disk. 
    high_command_set = set()
```

File: scripts/ci/service_name.py [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+
+
+def check():
+    az_cli = AzCli(cli_name='az',
+                   commands_loader_cls=MainCommandsLoader,
+                   invocation_cls=AzCliCommandInvoker,
+                   parser_cls=AzCliCommandParser,
+                   help_cls=AzCliHelp)
+    help_files = get_extension_help_files(az_cli)
+    # High command represents left most word in a command, e.g., vm, disk.
```
Comment (qwordy) [inline scripts/ci/service_name.py:96]:
```suggestion
    # High command represents left most word in a command, e.g., vm, disk.
```

**Pull request merged**